### PR TITLE
makes defibrillation grab ghost

### PR DESCRIPTION
**Pull request by mc-oofert**
## About The Pull Request

when paddles are being set up it pulls the ghost if not DNR into the body
ghost can still leave to cancel

## Why It's Good For The Game

so PVT dinglefart doesnt have to realize they have to reenter their body to get revived

## Changelog
:cl:
add: Defibrillation pulls the ghost into the body during the process
/:cl:

### CODE REVIEW

**Comment by Losenis** [comment]
based

**Comment by mc-oofert** [comment]
> Does this take you out of larva queue the way clicking the reenter body button does?

no clue but it literally just calls reenter_corpse() so i think ti does

**Comment by Al-1ce** [comment]
Being forced out of the larva queue is kinda bad but at the same time if you want that you should DNR.
I'm conflicted on this.

**Comment by mc-oofert** [comment]
> Sometimes people 50/50 on whether they want to go xeno or wait for a revive, or just queue up so that if their body does end up going cold they won't be at the back of the larva queue. I guess you could put in a check to not auto-enter body if someone is queued up? I can't really think of another way to handle this.

literally just set DNR

or just shrimply

join larva queue if you died
getting defibbed? thats good youre probably revivable and getting revived

**Comment by Losenis** [comment]
If you want to queue for larva then please fucking DNR because I'm not going to be ecstatic about dragging some dead guy's ass through 20 xenos and attempt to revive him because I saw him with defib timer remaining only to find out he'll ignore the 20 defib attempts and suddenly go perma because he went xenoid. Worst case scenario for this PR is enforcing basic human decency.

**Comment by XElectricX** [comment]
This man is a rising star, speed merge.

File: code/game/objects/items/defibrillator.dm [hunk 1]
```diff
@@ -140,6 +139,12 @@
 			return ghost
 	return null
 
+///GET IN DA BODY
+/mob/living/proc/grab_ghost()
+	var/mob/dead/observer/G = get_ghost()
+	. = G
+	if(G)
+		G.reenter_corpse()
```
Comment (Lumipharon) [inline code/game/objects/items/defibrillator.dm:147]: Do you really need a proc for this? Its just an extra proc call.

**Review by Lumipharon — CHANGES_REQUESTED** [review]

File: code/game/objects/items/defibrillator.dm (hunk 1 above)
Comment (mc-oofert) [reply code/game/objects/items/defibrillator.dm:147]: yeah

Comment (Lumipharon) [reply code/game/objects/items/defibrillator.dm:147]: just call reenter_corpse

Comment (mc-oofert) [reply code/game/objects/items/defibrillator.dm:147]: no

**Comment by Derrick2Electricboogaloo** [comment]
Yes please. One thing from TG we definitely needed.

**Review by Lumipharon — APPROVED** [review]

**Comment by BlueHNT** [comment]
I recommend adding a part with a conditional where it won't forcefully grab the ghost if you're in a larva queue but still ping you instead. I don't know how though.

**Comment by mc-oofert** [comment]
> I recommend adding a part with a conditional where it won't forcefully grab the ghost if you're in a larva queue but still ping you instead. I don't know how though.

if you scrolled up a little and read the comments...

**Pull request merged**